src/SubjectsHandler.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Queries for all subjects in the current queryEngine source
+ */
+export const SELECT_ALL_SUBJECTS = 'SELECT distinct ?thing WHERE { ?thing ?pred ?type }';
+export default class SubjectsHandler {
+  async *handle(pathData) {
+    // Retrieve the query engine
+    const { queryEngine } = pathData.settings;
+    if (!queryEngine)
+      throw new Error(`${pathData} has no queryEngine setting`);
+
+    // simply execute the query and yield the results
+    for await (const bindings of queryEngine.execute(SELECT_ALL_SUBJECTS))
+      yield this.extractTerm(bindings, pathData);
+  }
+
+  extractTerm(binding, pathData) {
```

Comment:
Code duplication of `ExecuteQueryHandler`, probably want to inherit from it.
Or, alternatively, we can change the `.sparql` handler instead, such that the `DISTINCT` query is generated there.

Reply:
updating `.sparql` was my initial approach, but it was becoming more complicated than I thought. I'll keep thinking about it though, that smells like a better solution in the long run